src/test/java/org/mvel2/tests/perftests/ConcurrentEvaluationWithContextTests.java

```diff
@@ -0,0 +1,222 @@
+package org.mvel2.tests.perftests;
+
+import java.io.Serializable;
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import junit.framework.TestCase;
+import org.junit.Test;
+import org.mvel2.MVEL;
+import org.mvel2.ParserContext;
+import org.mvel2.integration.VariableResolverFactory;
+import org.mvel2.integration.impl.IndexedVariableResolverFactory;
+import org.mvel2.optimizers.OptimizerFactory;
+import org.mvel2.tests.core.res.DefaultKnowledgeHelper;
+import org.mvel2.tests.core.res.NumberHolder;
+
+public class ConcurrentEvaluationWithContextTests extends TestCase {
+
+    private static final int LOOP = 100;
+
+    private String[] names = {"result", "$bus", "$title"};
+
+    private Serializable expression;
+
+    private void compileExpression() {
+
+        boolean allowNakedMethCall = MVEL.COMPILER_OPT_ALLOW_NAKED_METH_CALL;
+        boolean allowOverrideAllProphandling = MVEL.COMPILER_OPT_ALLOW_OVERRIDE_ALL_PROPHANDLING;
+        boolean allowResolveInnerclassesWithDotnotation = MVEL.COMPILER_OPT_ALLOW_RESOLVE_INNERCLASSES_WITH_DOTNOTATION;
+        boolean supportJavaStyleClassLiterals = MVEL.COMPILER_OPT_SUPPORT_JAVA_STYLE_CLASS_LITERALS;
+
+        try {
+            // mimic drools usage
+            MVEL.COMPILER_OPT_ALLOW_NAKED_METH_CALL = true;
+            MVEL.COMPILER_OPT_ALLOW_OVERRIDE_ALL_PROPHANDLING = true;
+            MVEL.COMPILER_OPT_ALLOW_RESOLVE_INNERCLASSES_WITH_DOTNOTATION = true;
+            MVEL.COMPILER_OPT_SUPPORT_JAVA_STYLE_CLASS_LITERALS = true;
+
+            ParserContext parserContext = ParserContext.create();
+            parserContext.setStrictTypeEnforcement(true);
+            parserContext.setStrongTyping(true);
+            parserContext.setIndexAllocation(true);
+
+            parserContext.addIndexedInput(names);
+            parserContext.addInput("result", List.class);
+            parserContext.addInput("$bus", Bus.class);
+            parserContext.addInput("$title", String.class);
+
+            expression = MVEL.compileExpression("result.add($bus.karaoke.dvd[$title].artist);", parserContext);
+
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
You might as well let it through by adding a `throws` clause. The test won't succeed if it gets here.

Reply:
Thank you for the advice! Fixed.